servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala

```diff
@@ -96,18 +96,11 @@ class Http4sServlet[F[_]](
       request: Request[F],
       bodyWriter: BodyWriter[F]): F[Unit] = {
     ctx.addListener(new AsyncTimeoutHandler(request, bodyWriter))
-    // TODO: This can probably be handled nicer with Effect
+    // Note: We're catching silly user errors in the lift => flatten.
```

Comment:
This is a case where I'd be happy to lecture about the silliness.

There is a similar construction in the blaze Http1ServerStage.

Reply:
the one in the `Http1ServerStage` is of the form:
```scala
  executionContext.execute(new Runnable {
          def run(): Unit =
            F.runAsync {
                try serviceFn(req)
                  .getOrElse(Response.notFound)
                  .handleErrorWith(serviceErrorHandler(req))
                catch serviceErrorHandler(req)
              } {
                case Right(resp) =>
                  IO(renderResponse(req, resp, cleanup))
                case Left(t) =>
                  IO(internalServerError(s"Error running route: $req", t, req, cleanup))
              }
              .unsafeRunSync()
        })
```

This is similar to writing `fs2.asyncUnsafeRunAsync(F)(callback)` but with the exception that `F.shift(ec)` submits a task onto the execution context.

Maybe im being a paranoid panda but while the ServerStage1 is uglier, it might be more efficient (In a sense, it's not submitting a useless task to evaluate the continuation) and I'm afraid of a tiny performance regression because of a change there.

Reply:
That's one of the hotspots in a ping test, which sadly is how people choose libraries.  (Shakes fist at naive reactions to Techempower.)  Okay, let's be conservative there, but the sanctimonious logging could be the same.